File: ios/CocoSleep/WidgetKitModule.m
#import <React/RCTBridgeModule.h>
#import "CocoSleep-Swift.h"

@interface CocoWidgetKitModule : NSObject <RCTBridgeModule>
@end

@implementation CocoWidgetKitModule

// Exposes as NativeModules.WidgetKit on the JS side
RCT_EXPORT_MODULE(WidgetKit)

RCT_EXPORT_METHOD(reloadAllTimelines) {
  [WidgetKitHelper reloadAll];
}

+ (BOOL)requiresMainQueueSetup {
  return NO;
}

@end

File: ios/CocoSleep/WidgetKitModule.swift
import Foundation
import WidgetKit

/// Exposed to ObjC so WidgetKitModule.m can call it.
@objc class WidgetKitHelper: NSObject {
  @objc static func reloadAll() {
    if #available(iOS 14.0, *) {
      WidgetCenter.shared.reloadAllTimelines()
    }
  }
}
